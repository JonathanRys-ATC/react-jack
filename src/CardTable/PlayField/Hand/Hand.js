import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Card from '../Card/Card'

export default class Hand extends Component {
    shouldComponentUpdate(nextProps) {
        if (this.props.faceUp === nextProps.faceUp &&
            this.props.hand === nextProps.hand) return false;
        return true;
    }

    render() {
        return (
            <div>
                {
                    this.props.hand.map((card, key) => {
                        return <Card key={`card-face-${key}`} card={card} faceUp={this.props.faceUp + key} />
                    })
                }
            </div>
        )
    }
}

Hand.propTypes = {
    faceUp: PropTypes.number.isRequired,
    hand: PropTypes.array.isRequired,
}
