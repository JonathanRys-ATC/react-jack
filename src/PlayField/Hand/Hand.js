import React from 'react'
import PropTypes from 'prop-types'

import Card from './Card/Card'

// export default class Hand extends Component {
const Hand = ({ hand, dealer }) => {
    return (
        hand.map((card, key) => <Card key={`card-face-${key}`} card={card} faceUp={dealer && !key ? false : true} />)
    )
}

Hand.propTypes = {
    dealer: PropTypes.bool,
    hand: PropTypes.array.isRequired,
}

export default Hand
