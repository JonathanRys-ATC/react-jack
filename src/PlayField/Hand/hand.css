.PlayField_Hand {
    min-height: 125px
}