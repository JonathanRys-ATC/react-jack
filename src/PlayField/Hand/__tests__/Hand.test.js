import React from 'react'
import { shallow } from 'enzyme'

import Hand from '../Hand'

const mockProps = { hand: ["SA", "SJ"], dealer: false }

describe("Test Hand component", () => {
    it("matches the snapshot", () => {
        const testProps = { ...mockProps }
        const component = shallow(<Hand {...testProps} />)
        expect(component).toMatchSnapshot()
    })

    it("matches the snapshot", () => {
        const testProps = { ...mockProps }
        testProps.dealer = true
        const component = shallow(<Hand {...testProps} />)
        expect(component).toMatchSnapshot()
    })
})