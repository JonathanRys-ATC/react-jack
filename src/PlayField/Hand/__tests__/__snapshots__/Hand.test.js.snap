// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hand component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hand
    dealer={false}
    hand={
      Array [
        "SA",
        "SJ",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Card
      card="SA"
      faceUp={false}
    />,
    <Card
      card="SJ"
      faceUp={false}
    />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Card
        card="SA"
        faceUp={false}
      />,
      <Card
        card="SJ"
        faceUp={false}
      />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test Hand component matches the snapshot 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hand
    dealer={true}
    hand={
      Array [
        "SA",
        "SJ",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Card
      card="SA"
      faceUp={true}
    />,
    <Card
      card="SJ"
      faceUp={false}
    />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Card
        card="SA"
        faceUp={true}
      />,
      <Card
        card="SJ"
        faceUp={false}
      />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
